core/node/notifications/event_processor.go

```diff
@@ -297,7 +298,11 @@ func (p *MessageToNotificationsProcessor) sendNotification(
 
 	var receivers []string
 	if channelID.Type() == shared.STREAM_DM_CHANNEL_BIN || channelID.Type() == shared.STREAM_GDM_CHANNEL_BIN {
-		receivers = members.ToSlice()
+		for _, member := range members.ToSlice() {
+			if member != sender.Hex() {
```

Comment:
`sender.Hex()` creates each time a new hex string while it is stable during the iteration. Do something like `senderHex := sender.Hex()` outside of the loop and do the comparison with that inside the loop.